src/test/javascript/spec/component/exam/participate/summary/exercises/quiz-exam-summary.component.spec.ts

```diff
@@ -0,0 +1,118 @@
+
+const shortAnswerQuestion = { id: 3, type: QuizQuestionType.SHORT_ANSWER } as ShortAnswerQuestion;
+const shortAnswerSpot = { id: 1, width: 1, spotNr: 1, question: shortAnswerQuestion, posX: 1, posY: 1, tempID: 1 } as ShortAnswerSpot;
+const shortAnswerSolution = { id: 1, text: 'solution', question: shortAnswerQuestion, posX: 1, posY: 1, tempID: 1 } as ShortAnswerSolution;
+const correctShortAnswerMapping = {
+    id: 1,
+    shortAnswerSolutionIndex: 1,
+    shortAnswerSpotIndex: 1,
+    solution: shortAnswerSolution,
+    spot: shortAnswerSpot,
+    question: shortAnswerQuestion,
+} as ShortAnswerMapping;
+shortAnswerQuestion.correctMappings = [correctShortAnswerMapping];
+shortAnswerQuestion.solutions = [shortAnswerSolution];
+shortAnswerQuestion.spots = [shortAnswerSpot];
+
+const studentParticipation = { id: 1 } as StudentParticipation;
+
+const multipleChoiceSubmittedAnswer = { id: 1, selectedOptions: [correctAnswerOption], quizQuestion: multipleChoiceQuestion, scoreInPoints: 1 } as MultipleChoiceSubmittedAnswer;
+const dragAndDropSubmittedAnswer = { id: 1, mappings: [correctDragAndDropMapping], quizQuestion: dragAndDropQuestion, scoreInPoints: 1 } as DragAndDropSubmittedAnswer;
+const shortAnswerSubmittedAnswer = { id: 1, mappings: [correctDragAndDropMapping], quizQuestion: shortAnswerQuestion, scoreInPoints: 1 } as ShortAnswerSubmittedAnswer;
+const shortAnswerSubmittedText = { id: 1, spot: shortAnswerSpot, text: 'solution', submittedAnswer: shortAnswerSubmittedAnswer } as ShortAnswerSubmittedText;
+shortAnswerSubmittedAnswer.submittedTexts = [shortAnswerSubmittedText];
+
+const submissionWithAnswers = {
+    id: 1,
+    submittedAnswers: [multipleChoiceSubmittedAnswer, dragAndDropSubmittedAnswer, shortAnswerSubmittedAnswer],
+    submitted: true,
+} as QuizSubmission;
+const exercise = { id: 1, studentParticipations: [studentParticipation], quizQuestions: [multipleChoiceQuestion, dragAndDropQuestion, shortAnswerQuestion] } as QuizExercise;
+
+describe('QuizExamSummaryComponent', () => {
+    let fixture: ComponentFixture<QuizExamSummaryComponent>;
+    let component: QuizExamSummaryComponent;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [HttpClientTestingModule, MockModule(ArtemisQuizQuestionTypesModule)],
+            declarations: [QuizExamSummaryComponent, MockPipe(TranslatePipe)],
+            providers: [MockProvider(TranslateService), MockProvider(QuizExerciseService), MockProvider(ArtemisServerDateService)],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(QuizExamSummaryComponent);
+                component = fixture.componentInstance;
+                component.exercise = exercise;
+                component.submission = { id: 2, submittedAnswers: [] };
+                component.resultsPublished = true;
+                component.exam = { id: 1 } as Exam;
```

Comment:
This is kind of overshadowed for test 1 because of line 101. If the idea behind it is that test 1 needs more exam information, you could move this into the beginning of test 2